### PR TITLE
Remove elmah table usage from gallery

### PR DESCRIPTION
**Pull request by drewgillies**
Partially addresses: https://github.com/NuGet/Engineering/issues/5480

We'll remove this table's usage as a part of the SDK migration effort.

### CODE REVIEW

**Comment by drewgillies** [comment]
FYI @Lanaparezanin

File: src/NuGetGallery/Web.config [hunk 1]
```diff
@@ -325,7 +274,7 @@
   -->
   <system.web>
     <httpCookies requireSSL="true" httpOnlyCookies="true"/>
-    <compilation debug="false" targetFramework="4.7.2">
+    <compilation debug="true" targetFramework="4.7.2">
```
Comment (erdembayar) [inline src/NuGetGallery/Web.config:277]: Should we revert this change?

Comment (blowdart) [reply src/NuGetGallery/Web.config:277]: Definitely - that should never be on in a production system, it gives away a bunch of information to attackers.

Comment (drewgillies) [reply src/NuGetGallery/Web.config:277]: +1. I'll get rid of this.

**Review by erdembayar — APPROVED** [review]
One less dependency, :)

**Comment by joelverhagen** [comment]
PR against `dev` please

File: src/NuGetGallery.Services/Diagnostics/SendErrorsToTelemetryAttribute.cs [hunk 2]
```diff
@@ -8,13 +8,10 @@
 namespace NuGetGallery.Diagnostics
 {
     [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = true)]
-    public sealed class SendErrorsToTelemetryAttribute
-        : ElmahHandleErrorAttribute
+    public sealed class SendErrorsToTelemetryAttribute : Attribute
```
Comment (joelverhagen) [inline src/NuGetGallery.Services/Diagnostics/SendErrorsToTelemetryAttribute.cs:11]: Does this still get invoked without Elmah installed?

File: src/NuGetGallery/App_Start/AppActivator.cs [hunk 3]
```diff
@@ -295,7 +293,7 @@ private static void BackgroundJobsPostStart(IAppConfiguration configuration)
                 {
                     RestartSchedulerOnFailure = true
                 };
-                _jobManager.Fail(e => ErrorLog.GetDefault(null).Log(new Error(e)));
+                _jobManager.Fail(e => { /* fail silently */ });
```
Comment (joelverhagen) [inline src/NuGetGallery/App_Start/AppActivator.cs:296]: It may be possible to use `Trace` here. I believe we have an App Insights trace listener. It might be worth having this "just in case".

**Comment by joelverhagen** [comment]
Love the clean-up!

**Review by joelverhagen — DISMISSED** [review]

**Review by erdembayar — APPROVED** [review]

**Comment by agr** [comment]
Should it include a change to admin index page to remove Elmah link?

File: src/NuGetGallery.Services/Diagnostics/SendErrorsToTelemetryAttribute.cs (hunk 2 above)
Comment (drewgillies) [reply src/NuGetGallery.Services/Diagnostics/SendErrorsToTelemetryAttribute.cs:11]: Good catch--I'll remove this.

**Review by agr — DISMISSED** [review]

**Comment by drewgillies** [comment]
Changing this back to a draft for now as this change breaks feature flags.

**Review by joelverhagen — APPROVED** [review]

**Comment by drewgillies** [comment]
@erdembayar @joelverhagen the file being unsigned broke the build, and it persists in the build to support Elmah migrations (backward and, in future, forward). I will review this in separate work to see if it can be removed without removing migration support. See https://github.com/NuGet/Engineering/issues/5495 for tracking.
